### ISSUE TITLE
Blank return statements do not raise an exception from the TS compiler.

### ISSUE THREAD

**Issue by aabtop** [aabtop/reify#21]
A situation arose where the code:

```ts
  build(): h.Region3 {
    return
    h.Difference3({
      a: h.Difference3({
        a: Box3(this.width, this.length, this.height),
        b: Box3(this.width - (this.wall_thickness * 2), this.length - (this.wall_thickness * 2), this.height - (this.wall_thickness * 2)),
      }),
      b: h.Transform3({ source: Box3(2, 2, 2), transform: h.Translate3([0, 0, 3]) })
    });
  }
```

Resulted in a segmentation fault.  What was happening was that the first `return` statement on its own line was being interpreted as a statement on its own, and somehow getting past the type checker as it is clearly not returning a h.Region3 here.

Perhaps there are some TypeScript compiler settings that would enable more strict checking of types here?

Attached is the full file that seg faults on `hypo bad_house.ts Main house`.

**Comment by aabtop** [comment]
Lots of discussion over at https://github.com/microsoft/TypeScript/issues/5916 about this.  Seems like we should enable `--strictNullChecks` and/or `--noImplicitReturns`.

**Comment by aabtop** [comment]
Fixed in e690817e020e9d94f61b14751281c1e70fb45e8a .
